### PR TITLE
Graviton processors cheaper (and more performant) than x86 closes #142

### LINKED ISSUE #142: Graviton processors cheaper (and more performant) than x86
**Is your feature request related to a problem? Please describe.**
It has shown that Graviton2 and graviton3 processors are more performant and cheaper than equivalent x86 processors

**Describe the solution you'd like**
Implement findings for non graviton processors and suggest graviton2 or 3 instead.
Should be suggested for Linux, Ubuntu, ECS, EKS, k8s, RDS, 
 - (not for windows machines)

**Describe alternatives you've considered**
None

**Additional context**
Here are some links to comparisons: 
- https://dzone.com/articles/everything-you-need-to-know-about-aws-graviton2 
- https://dev.to/leading-edje/faster-cheaper-aws-graviton-2-2b0c

### PR DESCRIPTION
**Pull request by khushboo9024**
### Checklist
- [ ] Issue(s) linked

### CODE REVIEW

File: controls/rds.sp [hunk 1]
```diff
@@ -211,3 +212,31 @@ control "rds_db_low_utilization" {
...
+        else 'alarm'
+      end as status,
+      case
+        when class like 'db.%g%.%' then title || ' is using Graviton2 processor.'
+        else title || ' is not using Graviton2 processor.'
+      end as reason
+      ${local.tag_dimensions_sql}
+      ${local.common_dimensions_sql}
+    from
+      aws_rds_db_instance
```
Comment (madhushreeray30) [inline controls/rds.sp:239]:
```suggestion
      aws_rds_db_instance;
```

File: controls/eks.sp [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+      when ami_type = 'CUSTOM%' and a.architecture <> 'arm_64' and a.platform = 'linux' then title || ' is not using Graviton2 processor.'
+      when ami_type not like 'AL2_%' then title || ' is not using linux platform.'
+      when ami_type = 'AL2_ARM_64' then title || ' is using Graviton2 processor.'
+      else title || ' is not using Graviton2 processor.'
+    end as reason
+    ${local.tag_dimensions_sql}
+    ${local.common_dimensions_sql}
+  from
+    aws_eks_node_group as g
+    left join ami_architecture as a on a.node_group_arn = g.arn
```
Comment (madhushreeray30) [inline controls/eks.sp:82]:
```suggestion
    left join ami_architecture as a on a.node_group_arn = g.arn;
```

File: controls/ecs.sp [hunk 3]
```diff
@@ -120,3 +121,33 @@ control "ecs_service_without_autoscaling" {
...
+      case
+        when i.platform = 'windows' then i.title || ' is windows type machine.'
+        when i.architecture = 'arm64' then i.title || ' is using Graviton2 processor.'
+        else i.title || ' is not using Graviton2 processor.'
+      end as reason
+      ${replace(local.tag_dimensions_qualifier_sql, "__QUALIFIER__", "c.")}
+      ${replace(local.common_dimensions_qualifier_sql, "__QUALIFIER__", "c.")}
+    from
+      aws_ecs_container_instance as c
+      left join aws_ec2_instance as i on c.ec2_instance_id = i.instance_id
```
Comment (madhushreeray30) [inline controls/ecs.sp:151]:
```suggestion
      left join aws_ec2_instance as i on c.ec2_instance_id = i.instance_id;
```

File: controls/ecs.sp [hunk 4]
```diff
@@ -22,7 +22,8 @@ benchmark "ecs" {
   documentation = file("./controls/docs/ecs.md")
   children = [
     control.ecs_cluster_low_utilization,
-    control.ecs_service_without_autoscaling
+    control.ecs_service_without_autoscaling,
+    control.ecs_cluster_container_instance_with_graviton
```
Comment (madhushreeray30) [inline controls/ecs.sp:26]: please sort it >
```
control.ecs_cluster_container_instance_with_graviton,
control.ecs_cluster_low_utilization,
control.ecs_service_without_autoscaling
```

**Comment by misraved** [comment]
Closes https://github.com/turbot/steampipe-mod-aws-thrifty/issues/142

**Review by misraved — APPROVED** [review]

**Pull request merged**